core/interop/src/loaders/hashmap.rs

```diff
@@ -0,0 +1,63 @@
+//! A `ModuleLoader` that loads modules from a `HashMap` based on the name.
+use rustc_hash::FxHashMap;
+
+use boa_engine::module::{ModuleLoader, Referrer};
+use boa_engine::{Context, JsNativeError, JsResult, JsString, Module};
+use boa_gc::GcRefCell;
+
+/// A `ModuleLoader` that loads modules from a `HashMap` based on the name.
+/// After registering modules, this loader will look for the exact name
+/// in its internal map to resolve.
+#[derive(Debug, Clone)]
+pub struct HashMapModuleLoader(GcRefCell<FxHashMap<JsString, Module>>);
+
+impl Default for HashMapModuleLoader {
+    fn default() -> Self {
+        Self(GcRefCell::new(FxHashMap::default()))
+    }
+}
+
+impl HashMapModuleLoader {
+    /// Creates an empty `HashMapModuleLoader`.
+    #[must_use]
+    pub fn new() -> Self {
```

Comment:
suggestion/issue: not sure that new is needed here if all it does is call `Default`

Reply:
Some people don't like to import default (it's not part of std prelude). I've seen this pattern in places, including the rust std: https://github.com/rust-lang/rust/blob/master/library/std/src/collections/hash/map.rs#L229-L234

I'm okay replacing it with `empty()` too if you want.